src/userdev/java/net/minecraftforge/gradle/userdev/jarjar/JarJarProjectExtension.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * ForgeGradle
+ * Copyright (C) 2018 Forge Development LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+package net.minecraftforge.gradle.userdev.jarjar;
+
+import groovy.lang.GroovyObjectSupport;
+import groovy.namespace.QName;
+import groovy.util.Node;
+import groovy.util.NodeList;
+import net.minecraftforge.gradle.userdev.DependencyManagementExtension;
+import net.minecraftforge.gradle.userdev.UserDevPlugin;
+import net.minecraftforge.gradle.userdev.dependency.DependencyFilter;
+import net.minecraftforge.gradle.userdev.tasks.JarJar;
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.*;
+import org.gradle.api.attributes.Attribute;
+import org.gradle.api.publish.maven.MavenArtifact;
+import org.gradle.api.publish.maven.MavenPom;
+import org.gradle.api.publish.maven.MavenPublication;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class JarJarProjectExtension extends GroovyObjectSupport {
+    public static final String EXTENSION_NAME = "jarJar";
+    private static final String MAVEN_POM_NAMESPACE = "{http://maven.apache.org/POM/4.0.0}";
+
+    private final Attribute<String> fixedJarJarVersionAttribute = Attribute.of("fixedJarJarVersion", String.class);
+    private final Attribute<String> jarJarRangeAttribute = Attribute.of("jarJarRange", String.class);
+
+    private final Project project;
+
+    public JarJarProjectExtension(final Project project) {
+        this.project = project;
+    }
+
+    public void enable() {
+        if (project.getTasks().findByPath(UserDevPlugin.JAR_JAR_TASK_NAME) != null) {
+            Objects.requireNonNull(project.getTasks().findByPath(UserDevPlugin.JAR_JAR_TASK_NAME)).setEnabled(true);
```

Comment:
I believe you should use getByPath, which will throw a nice exception saying what was missing. Also, what about other jar jar tasks in the configuration that aren't the default? Do people have to state enabled = true in those custom tasks? Seems a bit weird.

Reply:
Not needed. The default task is always registered and the null check above takes core of it.

And no if you create a custom task task it will be enabled by default. The UserDevPlugin creates the default task and sets that to disabled.

Reply:
I see. then just use a variable?

Reply:
That could work, hold on will do.